tests/FSharp.Compiler.ComponentTests/Conformance/MethodResolution/ParametersResolution.fs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.ComponentTests.Conformance.MethodResolution
+
+open Xunit
+open FSharp.Test.Compiler
+
+module ParametersResolution =
+
+    [<Fact>]
+    let ``Method with optional and out parameters resolves correctly`` () =
+        Fsx """
+open System.Runtime.InteropServices
+
+type Thing =
+    static member Do(_: outref<bool>, [<Optional; DefaultParameterValue(1)>]i: int) = true
+let _, _ = Thing.Do(i = 1)
+let _, _ = Thing.Do()
+        """
+        |> typecheck
```

Comment:
Also add tests that check the actual return values, and if possible maybe add IL code generation tests.  There's plenty of scope for type checking to work but actual code generation to be wrong for this sort of thing :)

Reply:
Good point

Reply:
Done, added both value checking and IL verification